Tools/LocomotionTool/LocomotionTool.cs

```diff
@@ -0,0 +1,559 @@
+		Vector3 m_StartPosition;
+		Vector3 m_StartMidPoint;
+		Vector3 m_StartDirection;
+		float m_StartYaw;
+
+		bool m_Rotating;
+		bool m_WasRotating;
+		bool m_Crawling;
+		Vector3 m_RayOriginStartPosition;
+		Vector3 m_RayOriginStartForward;
+		Vector3 m_RayOriginStartRight;
+		Quaternion m_RigStartRotation;
+		Vector3 m_RigStartPosition;
+		Vector3 m_CameraStartPosition;
+		Quaternion m_LastRotationDiff;
+
+		// Allow shared updater to check input values and consume controls
+		LocomotionInput m_LocomotionInput;
+
+		Camera m_MainCamera;
+		float m_OriginalNearClipPlane;
+		float m_OriginalFarClipPlane;
+
+		Toggle m_FlyToggle;
+		Toggle m_BlinkToggle;
+		bool m_BlockValueChangedListener;
+
+		public ActionMap actionMap { get { return m_BlinkActionMap; } }
+
+		public Transform rayOrigin { get; set; }
+
+		public Transform cameraRig { private get; set; }
+
+		public List<ILinkedObject> linkedObjects { private get; set; }
+
+		public GameObject settingsMenuItemPrefab { get { return m_SettingsMenuItemPrefab; } }
+
+		public GameObject settingsMenuItemInstance
+		{
+			set
+			{
+				foreach (var toggle in value.GetComponentsInChildren<Toggle>())
+				{
+					if (toggle.isOn)
+					{
+						m_FlyToggle = toggle;
+						toggle.onValueChanged.AddListener(isOn =>
+						{
+							if (m_BlockValueChangedListener)
+								return;
+
+							// m_Preferences on all instances refer
+							m_Preferences.blinkMode = !isOn;
+							foreach (LocomotionTool linkedObject in linkedObjects)
+							{
+								if (linkedObject != this)
+								{
+									linkedObject.m_BlockValueChangedListener = true;
+									//linkedObject.m_ToggleGroup.NotifyToggleOn(isOn ? m_FlyToggle : m_BlinkToggle);
+									// HACK: Toggle Group claims these toggles are not a part of the group
+									linkedObject.m_FlyToggle.isOn = isOn;
+									linkedObject.m_BlinkToggle.isOn = !isOn;
+									linkedObject.m_BlockValueChangedListener = false;
+								}
+							}
+						});
+					}
+					else
+					{
+						m_BlinkToggle = toggle;
+					}
+				}
+			}
+		}
+
+		void Start()
+		{
+			if (this.IsSharedUpdater(this) && m_Preferences == null)
+			{
+				m_Preferences = new Preferences();
+
+				// Share one preferences object across all instances
+				foreach (LocomotionTool locomotionTool in linkedObjects)
+				{
+					locomotionTool.m_Preferences = m_Preferences;
+				}
+			}
+
+			m_BlinkVisualsGO = ObjectUtils.Instantiate(m_BlinkVisualsPrefab, rayOrigin);
+			m_BlinkVisuals = m_BlinkVisualsGO.GetComponentInChildren<BlinkVisuals>();
+			m_BlinkVisuals.enabled = false;
+			m_BlinkVisualsGO.transform.parent = rayOrigin;
+			m_BlinkVisualsGO.transform.localPosition = Vector3.zero;
+			m_BlinkVisualsGO.transform.localRotation = Quaternion.identity;
+
+			m_MainCamera = CameraUtils.GetMainCamera();
+			m_OriginalNearClipPlane = m_MainCamera.nearClipPlane;
+			m_OriginalFarClipPlane = m_MainCamera.farClipPlane;
+
+			Shader.SetGlobalFloat(k_WorldScaleProperty, 1);
```

Comment:
Discussion result: This is only happening here.  If/when that changes we'll centralize these things to ISetViewerScale